fairseq/models/transformer/transformer_config.py

```diff
@@ -111,13 +111,13 @@ class TransformerConfig(FairseqDataclass):
         },
     )
     adaptive_input: bool = False
-    encoder: EncDecBaseConfig = EncDecBaseConfig()
+    encoder: EncDecBaseConfig = field(default_factory=EncDecBaseConfig)
     # TODO should really be in the encoder config
     max_source_positions: int = field(
         default=DEFAULT_MAX_SOURCE_POSITIONS,
         metadata={"help": "Maximum input length supported by the encoder"},
     )
-    decoder: DecoderConfig = DecoderConfig()
+    decoder: DecoderConfig = field(default_factory=EncDecBaseConfig)
```

Comment:
should be `field(default_factory=DecoderConfig()`

Reply:
> should be `field(default_factory=DecoderConfig)`

Sorry for the mistake.